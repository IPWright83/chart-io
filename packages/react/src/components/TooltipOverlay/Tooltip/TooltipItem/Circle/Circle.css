.chart-it.indicator-circle {
    width: 10px;
    height: 10px;
    margin-top: 2px;
    margin-right: 5px;
    border-radius: 50%;
}
