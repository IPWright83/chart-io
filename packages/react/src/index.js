export * from "./components";
export * from "./hoc";
export * from "./hooks";
export * from "./store";
export * from "./types";
export * from "./utils";
export * from "./constants";
export * from "./themes";
