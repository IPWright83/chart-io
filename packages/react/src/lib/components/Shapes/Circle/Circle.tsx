import type { IColor } from "@d3-chart/types";
import React from "react";

export interface ICircleProps {
    /**
     * The fill color of the Circle
     */
    fill: IColor;
}

export function Circle({ fill }: ICircleProps) {
    const style = {
        background: `${fill}`,
        width: 10,
        height: 10,
        marginTop: 4,
        marginLeft: 5,
        marginRight: 5,
        borderRadius: "50%",
        flexShrink: 0,
    };

    return <div className="d3-chart indicator-circle" style={style} />;
}
