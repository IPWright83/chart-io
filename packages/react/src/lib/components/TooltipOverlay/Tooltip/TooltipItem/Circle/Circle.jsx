import React from "react";
import PropTypes from "prop-types";

const Circle = ({ fill }) => {
    const style = {
        background: fill,
        width: 10,
        height: 10,
        marginTop: 4,
        marginLeft: 5,
        marginRight: 5,
        borderRadius: "50%",
    };

    return <div className="chart-it indicator-circle" style={style} />;
};

Circle.propTypes = {
    /**
     * The fill color of the Circle
     * @type {String}
     */
    fill: PropTypes.string.isRequired,
};

export { Circle };
