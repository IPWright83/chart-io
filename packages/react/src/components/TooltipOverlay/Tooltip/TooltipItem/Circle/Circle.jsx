import React from "react";
import PropTypes from "prop-types";

import "./Circle.css";

const Circle = ({ fill }) => {
    const style = {
        background: fill,
    };

    return <div className="chart-it indicator-circle" style={style} />;
};

Circle.propTypes = {
    /**
     * The fill color of the Circle
     * @type {String}
     */
    fill: PropTypes.string.isRequired,
};

export { Circle };
