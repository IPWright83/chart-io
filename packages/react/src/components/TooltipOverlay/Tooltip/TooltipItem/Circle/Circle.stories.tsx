import React from "react";
import { Circle } from "./Circle";

import mdx from "../TooltipItem.mdx";

export default {
    title: "Components/TooltipOverlay/Tooltip/TooltipItem/Circle",
    component: Circle,
    parameters: {
        docs: {
            page: mdx,
        },
    },
};

const CircleTemplate = (args) => {
    return <Circle fill={args.fill} />;
};

export const Default = CircleTemplate.bind({});
Default.args = {
    fill: "#4682B4",
};
