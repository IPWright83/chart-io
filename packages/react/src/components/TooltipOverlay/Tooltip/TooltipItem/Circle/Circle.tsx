import React from "react";
import PropTypes from "prop-types";

import type { IColor } from "../../../../../types";

import "./Circle.css";

export interface ICircleProps {
    /**
     * The fill color of the Circle
     * @type {String}
     */
    fill: IColor;
}

const Circle = ({ fill }: ICircleProps) => {
    const style = {
        background: fill,
    };

    return <div className="chart-it indicator-circle" style={style} />;
};

export { Circle };
