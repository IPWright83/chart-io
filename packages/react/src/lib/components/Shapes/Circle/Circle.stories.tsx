import React from "react";
import { Circle } from "./Circle";

export default {
  title: "Components/TooltipOverlay/Tooltip/TooltipItem/Circle",
  component: Circle,
};

const CircleTemplate = (args) => {
  return <Circle fill={args.fill} />;
};

export const Default = CircleTemplate.bind({});
Default.args = {
  fill: "steelblue",
};
