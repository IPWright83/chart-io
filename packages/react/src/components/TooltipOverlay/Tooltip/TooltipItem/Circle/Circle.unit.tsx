import React from "react";
import { render } from "@testing-library/react";

import { Circle } from "./Circle";

describe("Circle", () => {
    it("renders correctly", () => {
        const { asFragment } = render(<Circle fill="#4682B4" />);

        expect(asFragment()).toMatchSnapshot();
    });
});
