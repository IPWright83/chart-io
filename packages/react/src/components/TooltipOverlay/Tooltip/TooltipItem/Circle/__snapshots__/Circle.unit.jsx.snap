// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle renders correctly 1`] = `
<DocumentFragment>
  <div
    class="chart-it indicator-circle"
    style="background: steelblue;"
  />
</DocumentFragment>
`;
